desktop/src/main/java/bisq/desktop/main/market/offerbook/OfferBookChartView.java

```diff
@@ -421,9 +427,47 @@ private void updateChartData() {
 
         seriesBuy.getData().addAll(model.getBuyData());
         seriesSell.getData().addAll(model.getSellData());
+
+        // Before adding the points to the chart, set a custom node used for showing a label on hover
+        Arrays.asList(seriesBuy, seriesSell).forEach(series -> {
+            series.getData().forEach(dataPoint -> {
+                dataPoint.setNode(
+                        new HoveredLabelNode(model.getCurrencyCode(), dataPoint.getXValue(), dataPoint.getYValue())
+                );
+            });
+        });
+
+        // Add both series to the chart
         areaChart.getData().addAll(List.of(seriesBuy, seriesSell));
     }
 
+    /**
+     * A node which displays a label on hover, but is otherwise empty
+     *
+     * Based on https://gist.github.com/jewelsea/1422628
+     */
+    protected class HoveredLabelNode extends StackPane {
```

Comment:
I would move this Node to the seprate file and pass data via constructor (the price ->`xAxis.getTickLabelFormatter().toString(price)`. 

This file is quite long already and this is separable responsibility.